lib/seahorn/BvOpSem2MemRepr.cc

```diff
@@ -358,12 +370,50 @@ OpSemMemLambdaRepr::MemCpy(PtrTy dPtr, PtrTy sPtr, unsigned len,
       Expr readFromSrc = op::bind::fapp(srcMem.toExpr(), readPtrInSrc.toExpr());
       Expr readFromDst = op::bind::fapp(memRead.toExpr(), b0.toExpr());
 
-      Expr ite = boolop::lite(cmp, readFromSrc, readFromDst);
+      // -- body of the new lambda function
+      Expr body;
+      if (remainderBytes) {
+        LOG("opsem.lambda",
+            WARN << "memcpy of incomplete words. potential bottleneck.");
+        // -- if there are remainder bytes, stitch the last word together
+
+        // -- address of last word in destination is after the last word copied
+        PtrTy lastWordAddr =
+            m_memManager.ptrAdd(dPtr, lastByteToCopy + wordSzInBytes);
+        Expr isLastWordCmp = m_memManager.ptrEq(b0, lastWordAddr);
+
+        // -- after compare, B0 is the same as last address
+        Expr lastWordValDst = op::bind::fapp(memRead.toExpr(), b0.toExpr());
+        // -- readPtrInSrc is an address in src that is at the corresponding
+        // offset from B0
+        Expr lastWordValSrc =
+            op::bind::fapp(srcMem.toExpr(), readPtrInSrc.toExpr());
+
+        // -- compute the last word by taking chunks of source and destination
+        // -- words. source word comes first
+        unsigned wordSzInBits = wordSzInBytes * 8;
+        unsigned remainderBits = remainderBytes * 8;
+        auto &alu = m_ctx.alu();
+        Expr srcChunk =
+            alu.Extract({lastWordValSrc, wordSzInBits}, 0, remainderBits - 1);
```

Comment:
Please confirm that `end` is inclusive

i.e. if `remainderBits = 4` then bits 0, 1, 2, 3 will be extracted.

Reply:
yes. definitely inclusive. Forgot `-1` at first and z3 complained with type error.